### PR TITLE
Adds exclusive ranges to XRANGE/XREVRANGE

### PR DESCRIPTION
**Pull request by itamarhaber**
See https://github.com/redis/redis/pull/8072

Also simplifies iteration examples as a result.

### CODE REVIEW

File: commands/xrange.md [hunk 1]
```diff
@@ -67,6 +67,13 @@ Used in this way `XRANGE` works as a range query command to obtain entries
 in a specified time. This is very handy in order to access the history
 of past events in a stream.
 
+## Exclusive ranges
+
+The range is open (inclusive) by default, meaning that the reply can include
+entries with IDs matching the query's start and end intervals. It is possible
+to specify an open interval (exclusive) by prefixing the ID with the
+character `(`.
```
Comment (oranagra) [inline commands/xrange.md:75]:
```suggestion
The range is open (inclusive) by default, meaning that the reply can includes
entries with IDs matching the query's start and end intervals. It is possible
to specify a closed interval (exclusive) by prefixing the ID with the
character `(`.
```

File: commands/xrevrange.md [hunk 2]
```diff
@@ -14,54 +14,6 @@ enough to send:
 
     XREVRANGE somestream + - COUNT 1
 
-## Iterating with XREVRANGE
```
Comment (oranagra) [inline commands/xrevrange.md:17]: do you think maybe there's a reason to keep it for the benefit of redis 5.0 and 6.0 users?

File: commands/xrange.md [hunk 3]
```diff
@@ -67,6 +67,13 @@ Used in this way `XRANGE` works as a range query command to obtain entries
 in a specified time. This is very handy in order to access the history
 of past events in a stream.
 
+## Exclusive ranges
+
+The range is open (inclusive) by default, meaning that the reply can include
```
Comment (itamarhaber) [inline commands/xrange.md:72]:
```suggestion
The range is close (inclusive) by default, meaning that the reply can include
```

File: commands/xrange.md (hunk 1 above)
Comment (itamarhaber) [reply commands/xrange.md]: Oops - thanks.

File: commands/xrevrange.md (hunk 2 above)
Comment (itamarhaber) [reply commands/xrevrange.md:17]: Added back as a subsection in XRANGE

**Review by oranagra — APPROVED** [review]
it would have been nice to put all that details inside some collapsible block

<details>
<summary>pre-6.2 iteration</summary>

bla
bla
bla

</details>

but we do what we can, so i guess what you did is enough.

**Pull request merged**